mptt/managers.py

```diff
@@ -672,6 +672,79 @@ def partial_rebuild(self, tree_id):
 
         self._rebuild_helper(pks[0], 1, tree_id)
 
+    @delegate_manager
+    def bulk_load(self, data, target=None, position='last-child'):
+        """
+        Load a tree from a nested dictionary for bulk insert, returning an
+        array of records. Use to efficiently insert many nodes within a tree
+        without an expensive `rebuild`.
+
+        ::
+
+            records = MyModel.objects.bulk_load({
+                'id': 7,
+                'name': 'parent',
+                'children': [
+                    {
+                        'id': 8,
+                        'parent_id': 7,
+                        'name': 'child',
+                        'children': [
+                            {
+                                'id': 9,
+                                'parent_id': 8,
+                                'name': 'grandchild',
+                            }
+                        ]
+                    }
+                ]
+            })
+            MyModel.objects.bulk_create(records)
+
+        """
+        opts = self.model._mptt_meta
+        if target:
+            tree_id = target.tree_id
+            if position in ('left', 'right'):
+                level = getattr(target, opts.level_attr)
+                if position == 'left':
+                    cursor = getattr(target, opts.left_attr)
+                else:
+                    cursor = getattr(target, opts.right_attr) + 1
+            else:
+                level = getattr(target, opts.level_attr) + 1
+                if position == 'first-child':
+                    cursor = getattr(target, opts.left_attr) + 1
+                else:
+                    cursor = getattr(target, opts.right_attr)
+        else:
+            tree_id = self.tree_model._tree_manager._get_next_tree_id()
+            cursor = 1
+            level = 0
+
+        stack = []
+
+        def treeify(data, cursor=1, level=0):
```

Comment:
Actually the `treeify` function uses the `stack` and `opts` variables from the enclosing scope so it should stay as it.